pkg/replicatedapp/api.go

```diff
@@ -40,8 +44,37 @@ type LicenseData struct {
 	License      *kotsv1beta1.License
 }
 
+func GetPreferredChannelSlug() (string, error) {
```

Comment:
I would suggest using the existing `GetInstallationParams` function: https://github.com/replicatedhq/kots/blob/2dae2bf18b640b2360f85e669228804f5d582a8e/pkg/kotsutil/kots.go#L1144

Reply:
Example usage: https://github.com/replicatedhq/kots/blob/2dae2bf18b640b2360f85e669228804f5d582a8e/pkg/automation/automation.go#L238-L241

Reply:
Also, if you don't mind, can you just have that `GetInstallationParams` always use `kotsadmtypes.KotsadmConfigMap` and not have to accept it as a parameter?

Reply:
fyi - went ahead updated `RemoveAppVersionLabelFromInstallationParams` too (but easy to undo if I shouldn't have).